src/components/HeaderForm.js

```diff
@@ -527,6 +533,20 @@ class HeaderForm extends Component {
     if (tracks) {
       this.setState({ tracks: tracks });
       console.log("New tracks have been population");
+    } else if (isValidURL(this.state.bedFile)){
+      // Try to retrieve tracks if the current bed file is an URL
+      const json = await fetchAndParse(`${this.props.apiUrl}/getChunkTracks`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ bedURL: this.state.bedFile, chunk: chunk }),
+      });
+
+      // Replace tracks if request returns non-falsey value
+      if (json.tracks) {
+        this.setState({ tracks: json.tracks });
```

Comment:
It's possible that the `getChunkTracks` request comes back after the user has already swapped to a different BED region. We only want to apply the tracks from the server if the region we asked about is still the selected one.

I think we want to do something like a setState function here, so we can see the state that's current when the result arrives from the server:
```suggestion
        this.setState((laterState) => {
          if (laterState.region === coords) {
            // The user still has the same region selected, so apply the tracks we now have
            return { tracks: json.tracks };
          }
          // Otherwise, don't apply the downloaded tracks, because they are no longer relevant.
          // TODO: Save the downloaded tracks in case the user selects the region again?
        });
```